### PR TITLE
Avoid "ad" anywhere in the biuldId

### PR DESCRIPTION
**Pull request by shahzeb1**
In response to the comment created here:

https://github.com/zeit/next.js/pull/6851/files/a3d7fab3f181d310673af31b7ac14522856e219f#diff-76c9c0ae51cc93af6d31aae67d12a4c2R7

### CODE REVIEW

File: packages/next/build/generate-build-id.ts [hunk 1]
```diff
@@ -2,7 +2,11 @@ export async function generateBuildId (generate: () => string|null, fallback: ()
   let buildId = await generate()
   // If there's no buildId defined we'll fall back
   if (buildId === null) {
-    buildId = fallback()
+    // We also create a new buildId if it starts with `ad` to avoid false 
+    // positives with ad blockers
+    while (!buildId || buildId.includes('ad')) {
```
Comment (lfades) [inline packages/next/build/generate-build-id.ts]: Can this be a regex instead ? we want to avoid `AD` too

Comment (shahzeb1) [reply packages/next/build/generate-build-id.ts]: On it.

**Review by shahzeb1 — COMMENTED** [review]
Using regex (which is case insensitive) to avoid anything with "ad" or "AD" anywhere as a substring.

File: packages/next/build/generate-build-id.ts [hunk 2]
```diff
@@ -2,7 +2,11 @@ export async function generateBuildId (generate: () => string|null, fallback: ()
   let buildId = await generate()
   // If there's no buildId defined we'll fall back
   if (buildId === null) {
-    buildId = fallback()
+    // We also create a new buildId if it starts with `ad` to avoid false 
+    // positives with ad blockers
+    while (!buildId || /ad/i.test(buildId)) {
```
[the remark this answers is not in the document]
Comment (Magellol) [reply packages/next/build/generate-build-id.ts]: I was quickly reading about one of the adblocking extension and it looks like they're also looking for words like `banner` and `adv` for instance. `banner` seems unlikely to happen but maybe `adv` (advertising).

https://stackoverflow.com/questions/17255493/how-to-stop-adblock-plus-blocking-images-in-html-page

[the remark this answers is not in the document]
Comment (lfades) [reply packages/next/build/generate-build-id.ts]: @Magellol `adv` will still be matched by this regex.

**Comment by lfades** [comment]
@shahzeb1 Can you resolve the conflicts ? I'll merge after that. Thank you so much 💯

File: packages/next/build/generate-build-id.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (Magellol) [reply packages/next/build/generate-build-id.ts]: You are right. Disregard my comment.

**Review by Timer — APPROVED** [review]

**Pull request merged**